### DISCUSSION TITLE
Timeline

### DISCUSSION THREAD

**Discussion by XiaofengXie16** [apollographql/router#867] [General]
I noticed that the apollo team made the router to be open preview recently. I am wondering when is apollo/router going to be ready for production use? any rough timeline?

**Comment by abernix** [comment]
Thanks for your inquiry, @XiaofengXie16!  We believe that the Apollo Router is ready to handle production scale workloads today.  We are of course gathering feedback from users adopting the router while its in its Preview stage and using that feedback to evolve the feature set and the API, including the surface area of our extensions API.  Overall though, unlike the alpha phase where we had regular breaking changes, the changes we're making at this point are minimal and anything we do change is clearly communicated in our release notes.

None of this should necessarily preclude you from running the Router in production though.  If you're hesitant but you're already using Apollo Gateway 2.x today, you might consider experimenting with the Router in a parallel environment and seeing how it behaves.  Both Apollo Gateway 2.x and the Apollo Router both support Federation 2 which [recently went to GA](https://www.apollographql.com/blog/announcement/backend/apollo-federation-2-is-now-generally-available/) (general availability), so they should be equally capable in that regard, though the Apollo Router is tremendously faster from a performance standpoint.

Practically speaking, we're trying to get the Router to GA (general availability) in the next 4-6 weeks, but nothing can help us get there faster than getting users to try it out and deliver feedback about their results.  We'd love your help!  We don't have any future stages planned that are explicitly more "ready for production use" than where we are today, though we will — as previously noted — polish things up and make even stronger support commitments to the final GA releases than we would make to the preview releases. 

I hope this helps, but is there anything you'd like to know specifically that might encourage you to consider experimenting?  The Router's behavior and performance tends to be consistent so, if your experiments work, there shouldn't be a reason not to take things to a production environment.  We also intend on having a smooth transition between preview and GA; if you start with preview today, you should graduate smoothly into GA when the time comes in the coming weeks.

**Reply by XiaofengXie16** [reply]
That's good to know, thank you so much for sharing this info!

**Reply by DoumanAsh** [reply]
@abernix Out of curiosity is there tracking for removal of Node.js as dependency? While I'm successfully building embedded router, having fat V8 dependency is a bit of bother because you cannot really build it targeting musl right now

**Reply by abernix** [reply]
@DoumanAsh There's not an issue for it, but I can tell you it's (quite) important to the team.  Re-writing query planning itself in Rust is something we'd like to do (but a "large" lift and we're waiting on [`apollo-rs`](https://github.com/apollographql/apollo-rs) to get to a particular state of maturity where it vastly simplfies doing so.

Practically speaking there's no reason that V8 can't compile on Musl (Node.js literally distributes such binaries), but there's an upstream issue that hasn't got much attention preventing it from working without a substantial amount of tinkering — see https://github.com/denoland/rusty_v8/issues/49.  If that's something you'd be able to help push along upstream, that could pay benefits to your cause!  Curiously, is there a particular target constraint that is forcing you to use Musl rather than something else?

**Reply by DoumanAsh** [reply]
@abernix No, there is no particular requirement for musl, we initially based our infrastructure on musl because it is smaller. But switching to some slim variant of debian is not a big issue for us, so in worse case we just need to update our infrastructure for that.
I'll be keeping eye on `apollo-rs` thanks for mentioning it.
Glad to hear it is priority for team.
